### PR TITLE
when expressions with array results (doc and example)

### PR DESCRIPTION
**Pull request by pritidesai**
# Changes

Adding an example showcasing the usage of array results in `when` expressions, including indexing into an array results and array results with star notation. This usage includes both `input` and `values` components of `when` expressions.

Also, added doc with explicit syntax for `input` and `values`.

This is for the preparation of a potential promotion of https://github.com/tektoncd/pipeline/issues/5688.

Signed-off-by: pritidesai <pdesai@us.ibm.com>

/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Doc update for - How to refer to an array result in when expressions?
```

### CODE REVIEW

**Comment by pritidesai** [comment]
Hey @jerop, here are the changes (doc and example) for showcasing how to refer to array results in when expressions as promised in https://github.com/tektoncd/pipeline/pull/6055#issuecomment-1409227342, PTAL, thanks!

File: docs/pipelines.md [hunk 1]
```diff
@@ -590,9 +590,13 @@ tasks:
...
+| `input`    | Input for the `when` expression, defaults to an empty string if not provided.                              | * Static values e.g. `"ubuntu"`<br/> * Variables ([parameters](#specifying-parameters) or [results](#using-results)) e.g. `"$(params.image)"` or `"$(tasks.task1.results.image)"` or `"$(tasks.task1.results.array-results[1])"`                                                                                                                                                                                                                                               |
+| `operator` | `operator` represents an `input`'s relationship to a set of `values`, A valid `operator` must be provided. | `in` or `notin`                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
```
Comment (jerop) [inline docs/pipelines.md:597]:
```suggestion
| `operator` | `operator` represents an `input`'s relationship to a set of `values`, a valid `operator` must be provided. | `in` or `notin`                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
```

File: examples/v1beta1/pipelineruns/alpha/pipeline-with-when-expressions-using-array-results.yaml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        taskSpec:
+          results:
+            # a list of images in array-result-1
+            - name: array-results-1
+              type: array
+              description: The array results
+            # a list of images in array-result-2
+            - name: array-results-2
+              type: array
+              description: The array results
```
Comment (jerop) [inline examples/v1beta1/pipelineruns/alpha/pipeline-with-when-expressions-using-array-results.yaml:22]: looks like this result - `array-results-2` - is neither produced in this task nor used in subsequent tasks 🤔

File: docs/pipelines.md (hunk 1 above)
Comment (pritidesai) [reply docs/pipelines.md]: thank you @jerop, fixed it!

File: examples/v1beta1/pipelineruns/alpha/pipeline-with-when-expressions-using-array-results.yaml (hunk 2 above)
Comment (pritidesai) [reply examples/v1beta1/pipelineruns/alpha/pipeline-with-when-expressions-using-array-results.yaml]: yeah I just left it to make sure such result declaration does not impact here in any way, but on a second thought, may be not necessary here, deleted it.

**Review by jerop — APPROVED** [review]
thanks @pritidesai

**Comment by Yongxuanzhang** [comment]
/lgtm
Thanks! The example covers the use cases and it's a great example for me to learn!

**Pull request merged**